src/main/java/org/folio/inventory/dataimport/handlers/matching/loaders/AbstractLoader.java

```diff
@@ -68,16 +85,33 @@ public AbstractLoader(Vertx vertx) {
     return future;
   }
 
+  private boolean canProcessMultiMatchResult(DataImportEventPayload eventPayload) {
+    List<ProfileSnapshotWrapper> childProfiles = eventPayload.getCurrentNode().getChildSnapshotWrappers();
+    return isNotEmpty(childProfiles) && ReactTo.MATCH.equals(childProfiles.get(0).getReactTo())
+      && ContentType.MATCH_PROFILE.equals(childProfiles.get(0).getContentType());
+  }
+
   @Override
   public boolean isEligibleForEntityType(EntityType entityType) {
     return getEntityType() == entityType;
   }
 
+  protected String getConditionByMultiMatchResult(DataImportEventPayload eventPayload) {
+    String preparedIds = new JsonArray(eventPayload.getContext().remove(MULTI_MATCH_IDS))
+      .stream()
+      .map(Object::toString)
+      .collect(Collectors.joining(" OR "));
+
+    return format(" AND id == (%s)", preparedIds);
```

Comment:
it appears that this method is constructing some SQL is that correct? Thinking this should originate in some other class that deals with SQL statements if it does.

Reply:
this method returns CQL condition (not SQL) that is used by rest api of mod-inventory-storage